### ISSUE TITLE
If in project another chain router is used ONGR router doesn't work.

### ISSUE THREAD

**Issue by saimaz** [ongr-io/RouterBundle#86]
Some comments from gitter chat.

@funkdoobiest wrote:

Ok, seems like I found something
in ONGR\RouterBundle\DependencyInjection\Compiler\SetRouterPass

```php
<?php

namespace ONGR\RouterBundle\DependencyInjection\Compiler;

use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
use Symfony\Component\DependencyInjection\ContainerBuilder;
use Symfony\Component\DependencyInjection\Reference;

class SetRouterPass implements CompilerPassInterface
{
    public function process(ContainerBuilder $container)
    {
        //$container->setAlias('router', 'ongr_router.chain_router');
        $container
            ->getDefinition('ongr_router.elasticsearch_route_provider')
            ->addMethodCall(
                'setManager',
                [new Reference($container->getParameter('ongr_router.manager'))]
            );
    }
}
```

I commented first line in function and now it seems to work.
did not finished testing all things, but first look is good.
as I can see, some services in "ez platform" are getting router with "@router" param
and then it will get this alias from ongr which are without the "ez" routers
like here:

```yml

ezpublish.controller.content.download_redirection:
        class: %ezpublish.controller.content.download_redirection.class%
        arguments:
            - @ezpublish.api.service.content
            - @router
            - @ezpublish.route_reference.generator
        parent: ezpublish.controller.base

```

in other classes I can see that "ez" is getting all services tagged "router" and then everything worked fine
but all with `@router` failed

I hope my bad english was not to confusing
wrote this late at night

**Comment by saimaz** [comment]
ezpublish services already using chain router for their own logic. So this seems conflicts when we are trying to alias the Symfony router with another chain router which was already defined.

The first guess from out of nowhere is that maybe we should let to set alias for the end user. Cause by doing it strictly we just cause more problems. EZ publish is just one of many examples.

**Comment by saimaz** [comment]
@funkdoobiest, can you comment this line in your project vendor? 

https://github.com/ongr-io/RouterBundle/blob/73b64e2291f2a68e063960d32b5ac294358a6119/DependencyInjection/Compiler/SetRouterPass.php#L22

and take a look if that helps to sort things out ;)

**Comment by jmthoelken** [comment]
I've commented this line and now it's working :)

**Comment by saimaz** [comment]
Nice! Now I know the problem :)

**Comment by einorler** [comment]
Closed via #91
